homeassistant/components/advantage_air/climate.py

```diff
@@ -90,10 +91,14 @@ class AdvantageAirAC(AdvantageAirAcEntity, ClimateEntity):
     _attr_target_temperature_step = PRECISION_WHOLE
     _attr_max_temp = 32
     _attr_min_temp = 16
+    _attr_preset_modes = [ADVANTAGE_AIR_MYZONE]
 
-    def __init__(self, instance: AdvantageAirData, ac_key: str) -> None:
+    def __init__(
+        self, instance: AdvantageAirData, ac_key: str, config_entry: ConfigEntry
+    ) -> None:
         """Initialize an AdvantageAir AC unit."""
         super().__init__(instance, ac_key)
+        self._config_entry = config_entry
```

Comment:
This is already available as `instance.coordinator.config_entry` and `self.entity_platform.config_entry`

Reply:
Used instance.coordinator.config_entry instead, couldn't get self.entity_platform.config_entry or self.platform.config_entry to work.

Reply:
Used instance.coordinator.config_entry, couldn't get self.entity_platform.config_entry or self.platform.config_entry to work. Had to add a check that the config entry exists or else types failed.